src/Files.App/ViewModels/Properties/HashesViewModel.cs

```diff
@@ -120,6 +159,112 @@ private void ToggleIsEnabled(string? algorithm)
 			}
 		}
 
+		public bool CompareHash(string algorithm, string hashToCompare)
+		{
+			var hashInfoItem = Hashes.FirstOrDefault(x => x.Algorithm == algorithm);
+			if (hashInfoItem == null || hashInfoItem.HashValue == null)
+			{
+				return false;
+			}
+
+			return hashInfoItem.HashValue.Equals(hashToCompare, StringComparison.OrdinalIgnoreCase);
+		}
+
+		public async Task<bool> CompareFileAsync()
+		{
+			var result = CommonDialogService.Open_FileOpenDialog(MainWindow.Instance.WindowHandle, false, [], Environment.SpecialFolder.Desktop, out var toCompare);
+
+			if (!result)
+			{
+				return false;
+			}
+
+			var file = await StorageHelpers.ToStorageItem<BaseStorageFolder>(toCompare);
+
+			if (file is not null)
+			{
+				var selectedFileHash = await CalculateFileHashAsync(file.Path);
+				var compare = CompareHash("MD5", selectedFileHash);
+
+				return compare;
+			}
+
+			return false;
+		}
+
+		public string FindMatchingAlgorithm(string hash)
+		{
+			if (string.IsNullOrEmpty(hash))
+				throw new ArgumentNullException($"Invalid hash '{hash}' provided.");
+
+			foreach (var hashInfo in Hashes)
+			{
+				if (hashInfo.HashValue != null && hashInfo.HashValue.Equals(hash, StringComparison.OrdinalIgnoreCase))
+				{
+					return hashInfo.Algorithm;
+				}
+			}
+
+			return string.Empty;
+		}
+
+		public async Task<string> CalculateFileHashAsync(string filePath)
+		{
+			using (var stream = File.OpenRead(filePath))
+			{
+				using (var md5 = MD5.Create())
+				{
+					var hash = await Task.Run(() => md5.ComputeHash(stream));
+					return BitConverter.ToString(hash).Replace("-", "").ToLowerInvariant();
+				}
+			}
+		}
+
+		private void OnHashInputTextChanged()
+		{
+			string? matchingAlgorithm = null;
+
+			try
+			{
+				matchingAlgorithm = FindMatchingAlgorithm(HashInput);
+			}
+			catch (ArgumentNullException)
+			{
+				return;
+			}
+
+			if (string.IsNullOrEmpty(matchingAlgorithm))
+			{
+				InfoBarSeverity = InfoBarSeverity.Error;
+				InfoBarTitle = Strings.HashesDoNotMatch.GetLocalizedResource();
+			}
+			else
+			{
+				InfoBarSeverity = InfoBarSeverity.Success;
+				InfoBarTitle = string.Format(Strings.HashesMatch.GetLocalizedResource(), matchingAlgorithm);
+			}
+
+			IsInfoBarOpen = true;
+		}
+
+		private async Task OnCompareFileAsync()
+		{
+			var result = await CompareFileAsync();
+
+			if (result)
+			{
+				InfoBarSeverity = InfoBarSeverity.Success;
+				InfoBarTitle = Strings.HashesMatch.GetLocalizedResource();
+			}
+			else
+			{
+				InfoBarSeverity = InfoBarSeverity.Error;
+				InfoBarTitle = Strings.HashesDoNotMatch.GetLocalizedResource();
```

Comment:
TODO: Replace with proper strings

Reply:
What do you have in mind?

Reply:
"Files match" and "Files do not match"
Currently it displays "Matches {0} " and "Hashes don't match" since I'm reusing the previous strings